### DISCUSSION TITLE
ConvertUsing => Convert upgrade question

### DISCUSSION THREAD

**Discussion by GaryWayneSmith** [JoshClose/CsvHelper#1791] [General]
I have the code block below that has been working well for some time.  Updating to the latest breaks this section.  Reading the breaking changes I see that I need to use Convert.  In the first line that can be handled easily, in the second line there is a dependency on a value from the current object that's other than .Value.

Any advice on how I would fix this to put it inline with the current release?  I have about 60 such conversion in the code (not including normal custom conversions that are straight forward).

`
Map(m => m.InStock).Name("Availability").ConvertUsing(m => m.InStock ? "In Stock" : "Out of Stock");

Map(m => m.Inventory).Name("Quantity").ConvertUsing(m => m.InStock ? m.Inventory.ToString() : "0");
`

**Comment by GaryWayneSmith** [comment]
For some reason I was missing that Value is an object.  When inspecting it earlier it appeared that VS showed it as a string.

```
Map(m => m.InStock).Name("Availability").Convert(m => m.Value.InStock ? "In Stock" : "Out of Stock");
Map(m => m.Inventory).Name("Quantity").Convert(m => m.Value.InStock? m.Value.Inventory.ToString() : "0");
```

**Reply by JoshClose** [reply]
It's a single `args` object now, instead of several parameters.

**Reply by GaryWayneSmith** [reply]
I was able to solve all of the refactoring issues but this one.  Is there any substitute for the RawRecord as shown below?  I think the intent was the return the entire row in that field to be saved as raw text in a database.

```
old:
Map(m => m.ImportData).ConvertUsing(row => row.Context.RawRecord);

Tried:
Map(m => m.ImportData).Convert(m => m.Row.Context.RawRecord);

```

**Comment by JoshClose** [comment]
`Row.Parser.RawRecord`

**Reply by GaryWayneSmith** [reply]
Thank you for the follow up.  That should work for me.
